To: vim_dev@googlegroups.com
Subject: Patch 7.3.374
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.374
Problem:    ++encoding does not work properly.
Solution:   Recognize ++encoding before ++enc. (Charles Cooper)
Files:	    src/ex_docmd.c


*** ../vim-7.3.373/src/ex_docmd.c	2011-10-20 21:58:20.000000000 +0200
--- src/ex_docmd.c	2011-12-14 14:46:05.000000000 +0100
***************
*** 4840,4851 ****
  #ifdef FEAT_MBYTE
      else if (STRNCMP(arg, "enc", 3) == 0)
      {
! 	arg += 3;
! 	pp = &eap->force_enc;
!     }
!     else if (STRNCMP(arg, "encoding", 8) == 0)
!     {
! 	arg += 8;
  	pp = &eap->force_enc;
      }
      else if (STRNCMP(arg, "bad", 3) == 0)
--- 4840,4849 ----
  #ifdef FEAT_MBYTE
      else if (STRNCMP(arg, "enc", 3) == 0)
      {
! 	if (STRNCMP(arg, "encoding", 8) == 0)
! 	    arg += 8;
! 	else
! 	    arg += 3;
  	pp = &eap->force_enc;
      }
      else if (STRNCMP(arg, "bad", 3) == 0)
*** ../vim-7.3.373/src/version.c	2011-12-14 14:15:12.000000000 +0100
--- src/version.c	2011-12-14 14:44:01.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     374,
  /**/

-- 
Linux is just like a wigwam: no Windows, no Gates and an Apache inside.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
